test/system/src/main/java/io/pravega/test/system/framework/services/docker/DockerBasedService.java

```diff
@@ -72,10 +72,30 @@ public String getID() {
 
     @Override
     public boolean isRunning() {
-        boolean value = false;
+        try {
+            // List the service with filter 'serviceName'
+            Service.Criteria servicecriteria = Service.Criteria.builder().serviceName(serviceName).build();
+            List<Service> serviceList = Exceptions.handleInterruptedCall(
+                    () -> dockerClient.listServices(servicecriteria));
+            log.info("Service list size {}", serviceList.size());
+            if (!serviceList.isEmpty()) {
+                long replicas = Exceptions.handleInterruptedCall(
+                        () -> dockerClient.inspectService(serviceList.get(0).id()).spec().mode().replicated().replicas());
+                log.info("Replicas {}", replicas);
```

Comment:
Fetching the replica count can also be refactored to a separate method.

Reply:
Fixed